### PR TITLE
[M10-400] Fixed Ogury SDK Integration

### PR DESCRIPTION
**Pull request by carlosalcala-freestar**
**[M10-364](https://freestar.atlassian.net/browse/M10-364)**
**[M10-400](https://freestar.atlassian.net/browse/M10-400)**
**[M10-401](https://freestar.atlassian.net/browse/M10-401)**

## Description, Motivation and Context

**What is the goal of ticket?  What are the acceptance criteria?**

Integrate Ogury SDK - Banners
Integrate Ogury SDK - Interstitials
Integrate Ogury SDK - Rewards

## How Has This Been Tested? 

Manually tested App Build 

## Does this change require updates outside of the code, like updating the documentation, adding a feature flag, or adding fields to a DB table? If so, was this done?

N/A

## Any tickets that this is dependent upon?

N/A

## Should reviewers manually test your changes?

No

### Evidence

IMG_0423
IMG_0424
IMG_0425

### CODE REVIEW

File: FreestarAds-Ogury/Source/objc/OguryBannerMediator.m [hunk 1]
```diff
@@ -38,20 +37,27 @@ - (BOOL)canShowBannerAd {
...
--(void)loadBannerAd {
+- (void)loadInlineInviewAd {
     FSTRLog(@"OGURY: loadBannerAd");
 
-    self.ad = [[OguryBannerAd alloc] initWithAdUnitId:[self placementId]];
-    [self.ad loadWithSize:self.requestedSize];
+    FSTRLog(@"OGURY: placement_id %@", self.mPartner.placement_id);
+    FSTRLog(@"OGURY: adunitId %@", [self.mPartner adunitId]);
+
+    self.ad = [[OguryBannerAd alloc] initWithAdUnitId:[self.mPartner adunitId]];
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryBannerMediator.m:61]: Either add a nil check here to ensure you don't pass nil then call _partnerAdLoadFailed:_ or check for nil adunit id then call _partnerAdLoadFailed:_ and return early.

File: FreestarAds-Ogury/Source/objc/OguryInterstitialMediator.m [hunk 2]
```diff
@@ -27,17 +27,16 @@ - (BOOL)canShowInterstitialAd {
...
 - (void)loadInterstitialAd {
     FSTRLog(@"OGURY: loadInterstitialAd");
     [self resetInterstitialAd];
 
-    self.ad = [[OguryInterstitialAd alloc] initWithAdUnitId:[self placementId]];
-    [self.ad load];
+    FSTRLog(@"OGURY: placement_id %@", self.mPartner.placement_id);
+    FSTRLog(@"OGURY: adunitId %@", [self.mPartner adunitId]);
+
+    self.ad = [[OguryInterstitialAd alloc] initWithAdUnitId:[self.mPartner adunitId]];
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryInterstitialMediator.m:41]: Same here, should check adunit id for nil and if so, call _partnerAdLoadFailed:_

File: FreestarAds-Ogury/Source/objc/OguryRewardMediator.m [hunk 3]
```diff
@@ -27,17 +27,16 @@ - (BOOL)canShowRewardAd {
...
 - (void)loadRewardAd {
     FSTRLog(@"OGURY: loadRewardAd");
     [self resetRewardAd];
 
-    self.ad = [[OguryOptinVideoAd alloc] initWithAdUnitId:[self placementId]];
-    [self.ad load];
+    FSTRLog(@"OGURY: placement_id %@", self.mPartner.placement_id);
+    FSTRLog(@"OGURY: adunitId %@", [self.mPartner adunitId]);
+
+    self.ad = [[OguryOptinVideoAd alloc] initWithAdUnitId:[self.mPartner adunitId]];
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryRewardMediator.m:41]: Same here, should check adunit id for nil and if so, call partnerAdLoadFailed:

**Review by maddapper — COMMENTED** [review]
Optional, but you may want to add nil checks to ensure adunit id is valid before calling load.  That issue seems to have caused issue for you in the past...

**Review by maddapper — APPROVED** [review]
Optional, but you may want to add nil checks to ensure adunit id is valid before calling load. That issue seems to have caused issue for you in the past...

File: FreestarAds-Ogury/Source/objc/OguryBannerMediator.m (hunk 1 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryBannerMediator.m:61]: Fixed

File: FreestarAds-Ogury/Source/objc/OguryInterstitialMediator.m (hunk 2 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryInterstitialMediator.m:41]: Fixed

File: FreestarAds-Ogury/Source/objc/OguryRewardMediator.m (hunk 3 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryRewardMediator.m:41]: Fixed

**Review by maddapper — APPROVED** [review]

**Pull request merged**